content/post/azure-monitor-terraform.md

```diff
@@ -0,0 +1,230 @@
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_log_analytics_workspace" "example" {
+  name                = "acctest-01"
+  location            = azurerm_resource_group.example.location
+  resource_group_name = azurerm_resource_group.example.name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+
+resource "azurerm_log_analytics_workspace_table" "container_logs" {
+  name         = "ContainerLog"
+  workspace_id = azurerm_log_analytics_workspace.example.id
+
+  plan              = "Basic" # or "Analytics"
+  retention_in_days = 7 # per docs, setting to null defaults to workspace default
+}
+```
+
+
+
+The resource azurerm_log_analytics_workspace_table has since been deprecated. Currently there is no way to create the Log Analytics Workspace Table using the azurerm provider.
+
+The Workaround
+This is done by using the AzAPI to create the Log Analytics Workspace Table. The schema of the table should match the format of your logs.
+
+To enable the AzAPI you have to include it in your terraform configuration as follows:
+
+```hcl
+terraform {
+required_providers {
+    azapi = {
+      source  = "Azure/azapi"
+      version = "~> 1.8.0"
+    }
+  }
+}
+resource "azapi_resource" "data_collection_logs_table" {
+  name      = "Example_CL"
+  parent_id = azurerm_log_analytics_workspace.example.id
+  type      = "Microsoft.OperationalInsights/workspaces/tables@2022-10-01"
+  body = jsonencode(
+    {
+      "properties" : {
+        "schema" : {
+          "name" : "Example_CL",
+          "columns" : [
+            {
+              "name" : "TimeGenerated",
+              "type" : "datetime",
+              "description" : "The time at which the data was generated"
+            },
+            {
+              "name" : "Level",
+              "type" : "string",
+              "description" : "Log level"
+            },
+            {
+              "name" : "Logger",
+              "type" : "string",
+              "description" : "Logger name"
+            },
+            {
+              "name" : "Context",
+              "type" : "string",
+              "description" : "Context of the log line"
+            },
+            {
+              "name" : "Message",
+              "type" : "string",
+              "description" : "Log message"
+            },
+            {
+              "name" : "AdditionalContext",
+              "type" : "string",
+              "description" : "Additional context of the log line"
+            }
+          ]
+        },
+        "retentionInDays" : 30,
+        "totalRetentionInDays" : 30
+      }
+    }
+  )
+}
+```
+
+This HCL block will create the table in the associated Log Analytics Workspace using the defined schema. Details of schema options can be found in the documentation.
+
+The documentation for this can be found here: https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep
```

Comment:
Don't

> The documentation for this can be found here: https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep

Do

> The documentation for this can be found [here](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep)